ports/espressif/common-hal/pulseio/PulseIn.c

```diff
@@ -74,14 +74,20 @@ static bool _done_callback(rmt_channel_handle_t rx_chan,
 
 void common_hal_pulseio_pulsein_construct(pulseio_pulsein_obj_t *self, const mcu_pin_obj_t *pin,
     uint16_t maxlen, bool idle_state) {
+
+    // Only use dma when necessary, since dma-rmt might not be available.
+    // If dma is necessary but not available, this will raise an error below.
+    bool use_dma = maxlen > 128;
+
     self->buffer = (uint16_t *)m_malloc_without_collect(maxlen * sizeof(uint16_t));
     if (self->buffer == NULL) {
         m_malloc_fail(maxlen * sizeof(uint16_t));
     }
     // We add one to the maxlen version to ensure that two symbols at lease are
     // captured because we may skip the first portion of a symbol.
-    self->raw_symbols_size = MIN(64, maxlen / 2 + 1) * sizeof(rmt_symbol_word_t);
-    self->raw_symbols = (rmt_symbol_word_t *)m_malloc_without_collect(self->raw_symbols_size);
+    self->raw_symbols_size = (maxlen / 2 + 1) * sizeof(rmt_symbol_word_t);
+    // RMT DMA mode cannot access PSRAM -> ensure raw_symbols is in internal ram
+    self->raw_symbols = (rmt_symbol_word_t *)port_malloc(self->raw_symbols_size, use_dma);
```

Comment:
Since `port_malloc()` allocates outside of the CircuitPython heap, I think you're going to have a storage leak here if the program stops and you restart the VM. But since `PulseIn` has a finaliser, you could add a `port_free()` for `raw_symbols` to `common_hal_pulseio_pulsein_deinit()` .